lib/generators/shopify_app/app_proxy_controller/app_proxy_controller_generator.rb

```diff
@@ -0,0 +1,25 @@
+require 'rails/generators/base'
+
+module ShopifyApp
+  module Generators
+    class AppProxyControllerGenerator < Rails::Generators::Base
+      source_root File.expand_path('../templates', __FILE__)
+
+      def create_app_proxy_controller
+        template 'app_proxy_controller.rb', 'app/controllers/app_proxy_controller.rb'
+      end
+
+      def create_app_proxy_index_view
+        copy_file 'index.html.erb', 'app/views/app_proxy/index.html.erb'
+      end
+
+      def add_app_proxy_route
+        inject_into_file(
```

Comment:
I had a closer look at the link you provided.
the route method itself uses `inject_into_file`

``` ruby
def route(routing_code)
  log :route, routing_code
  sentinel = /\.routes\.draw do\s*\n/m

  in_root do
    inject_into_file 'config/routes.rb', "  #{routing_code}\n", { after: sentinel, verbose: false, force: false }
  end
end
```